unstoppable-ios-app/domains-manager-ios/Services/WalletConnect/WalletConnectService_V2/WalletConnectServiceV2.swift

```diff
@@ -700,22 +701,26 @@ extension WalletConnectServiceV2: WalletConnectV2RequestHandlingServiceProtocol
                 let sig = WCAnyCodable(sigString)
                 Debugger.printInfo(topic: .WalletConnectV2, "Successfully signed TX via external wallet: \(udWallet.address)")
                 return .response(sig)
+                
+            case .mpc: print("sign with mpc")
+                return .error(.internalError) // TODO: mpc
+                
+            default:  // locally verified wallet
+                guard let privKeyString = udWallet.getPrivateKey() else {
+                    Debugger.printFailure("No private key in \(udWallet)", critical: true)
+                    throw WalletConnectRequestError.failedToGetPrivateKey
+                }
+                
+                let privateKey = try EthereumPrivateKey(hexPrivateKey: privKeyString)
+                
+                let chainId = EthereumQuantity(quantity: BigUInt(chainIdInt))
+                
+                let signedTx = try completedTx.sign(with: privateKey, chainId: chainId)
+                let (r, s, v) = (signedTx.r, signedTx.s, signedTx.v)
+                let signature = r.hex() + s.hex().dropFirst(2) + String(v.quantity, radix: 16)
+                
+                return .response(WCAnyCodable(signature))
```

Comment:
I'm not sure but it looks like a code duplicity. There's already a sign function in the UDWallet which already handle self.type and run proper flow

Reply:
UDWallet has methods to sign messages, but this code signs tx